server/chat-plugins/usersearch.tsx

```diff
@@ -7,76 +7,101 @@ export const nameList = new Set<string>(JSON.parse(
 const ONLINE_SYMBOL = ` \u25C9 `;
 const OFFLINE_SYMBOL = ` \u25CC `;
 
-function getPunishmentHTML(userid: ID, target: string) {
-	return [
-		'Forcerename', 'Namelock', 'Weeknamelock',
-	].map((cmd) => (
-		`<button class="button" name="send" value="/msgroom staff,/${toID(cmd)} ${userid}&#10;/uspage ${target}">${cmd}</button>`
-	)).join(' | ');
-}
-
-function searchUsernames(target: string, page = false) {
-	const results: {offline: string[], online: string[]} = {
-		offline: [],
-		online: [],
-	};
-	for (const curUser of Users.users.values()) {
-		if (!curUser.id.includes(target) || curUser.id.startsWith('guest')) continue;
-		if (Punishments.isGlobalBanned(curUser)) continue;
-		if (curUser.connected) {
-			results.online.push(`${!page ? ONLINE_SYMBOL : ''} ${curUser.name}`);
-		} else {
-			results.offline.push(`${!page ? OFFLINE_SYMBOL : ''} ${curUser.name}`);
+class PunishmentHTML extends Chat.JSX.Component<{userid: ID, target: string}> {
+	render() {
+		const {userid, target} = {...this.props};
+		const buf = [];
+		for (const cmdName of ['Forcerename', 'Namelock', 'Weeknamelock']) {
+			// We have to use dangerouslySetInnerHTML here because otherwise the `value`
+			// property of the button tag is auto escaped, making &#10; into &amp;#10;
```

Comment:
Why not just unescape it? Use `\n` instead of `&#10;`? Would that work?

Reply:
I tried that, it doesn't work.